docs/developer/template_switches.asciidoc

```diff
@@ -0,0 +1,119 @@
+// to display images directly on GitHub
+ifdef::env-github[]
+:imagesdir: ../images
+endif::[]
+
+== Creating a new Switch via a Template
+
+To create a new Switch Template you must create a template file in the /usr/local/pf/lib/pf/Switch directory.
+The file must have the following pattern /usr/local/pf/lib/pf/Switch/<Vendor>/<SwitchName>.def
+
+For example to create a template for the vendor "Cyberdyne" and switch "Switchinator 800".
+The file name is /usr/local/pf/lib/pf/Switch/Cyberdyne/Switchinator800.def
+The file name must must only consist of alphanumeric characters and under scores and must begin with a letter.
+
+The file consist of parameter names and their values.
+
+=== Required Parameters
+
+* description      - The description of the switch.
+* radiusDisconnect - The RADIUS disconnect methods to use. Must be one of the following values coa|disconnect|coaOrdisconnect .
+
+=== RADIUS scope Parameters
+
+* acceptVlan - Attributes for accept vlan scope
+* acceptRole - Attributes for accept role scope.
+* reject     - Attributes for rejection scope.
+* disconnect - Attributes for disconnect scope (required if radiusDisconnect is 'disconnect' or 'coaOrdisconnect')
+* coa        - Attributes for CoA scope (required if radiusDisconnect is 'coa' or 'coaOrdisconnect')
+
+=== Comments
+
+The line of a comment must begin with a '#'
+For example:
+....
+# This is a comment
+....
+
+
+=== Defining RADIUS Attributes.
+
+scopeName = <<EOT
+Attribute-Name1 = value1
+Attribute-Name2 = value2
+EOT
+
+RADIUS Vendor Attributes for CoA
+scopeName = <<EOT
+VendorName:Attribute-Name1 = value1
+VendorName:Attribute-Name2 = value2
+EOT
+
+=== Dynamic RADIUS Attribute Value Syntax
+
+Some values depends on the context of the current request.
+So here is mini templating language to format values.
+
+Text replacement:  $name
+Embedded text replacement:  start-${name}-end
+Functions: ${f1("",$var, f2())}
+
+==== Available variables for RADIUS scope
```

Comment:
It would be useful to have access to the switch, node_info and other args part of the AAA context in here so we can do fancy stuff if we'd like

Reply:
Done